packages/engine-server/src/workspace/service.ts

```diff
@@ -469,6 +473,37 @@ export class WorkspaceService {
     return ws;
   }
 
+  static async createEmptyWorkspace(opts: WorkspaceServiceCreateOpts) {
+    const { wsRoot, vaults } = opts;
+    const ws = new WorkspaceService({ wsRoot });
+    fs.ensureDirSync(wsRoot);
+    // this creates `dendron.yml`
+    ws.createConfig();
+    // add gitignore
+    const gitIgnore = path.join(wsRoot, ".gitignore");
+    fs.writeFileSync(
+      gitIgnore,
+      ["node_modules", ".dendron.*", "build", "\n"].join("\n"),
+      { encoding: "utf8" }
+    );
+    if (opts.createCodeWorkspace) {
+      WorkspaceConfig.write(wsRoot, vaults);
+    }
+    await _.reduce(
+      vaults,
+      async (prev, vault) => {
+        await prev;
+        return ws.createVault({ vault });
+      },
+      Promise.resolve()
+    );
+    // check if this is the first workspace created
+    if (_.isUndefined(MetadataService.instance().getMeta().firstWsInitialize)) {
```

Comment:
since this is used in `createWorkspace` can we move this into a `postWorkspaceCreateHook` method? in the future, we could expose this to plugins to add their own hooks

Reply:
same comment above; this function will be removed.